Competition2018/src/main/java/competition/subsystems/drive/commands/DriveForDistanceCommand.java

```diff
@@ -0,0 +1,111 @@
+package competition.subsystems.drive.commands;
+import com.google.inject.Inject;
+
+import xbot.common.command.BaseCommand;
+import xbot.common.logging.RobotAssertionManager;
+import xbot.common.math.ContiguousHeading;
+import xbot.common.math.PIDManager;
+import xbot.common.math.PIDFactory;
+import xbot.common.properties.DoubleProperty;
+import xbot.common.properties.XPropertyManager;
+import competition.subsystems.drive.DriveSubsystem;
+import competition.subsystems.pose.PoseSubsystem;
+
+public class DriveForDistanceCommand extends BaseCommand {
+    
+    private final PIDManager travelManager;
+    private final PIDManager headingDrivePid;
+    private final PoseSubsystem poseSubsystem;
+    private final DriveSubsystem drive;
+    
+    public double leftPower;
+    public double rightPower;
+    private double deltaDistance;
+    private double targetDistance;
+    private double previousPositionInches;
+    public final double defaultPValue = 1/80d;
+    
+    private DoubleProperty deltaDistanceProp;
+    private ContiguousHeading targetHeading;
+    
+    @Inject
+    public DriveForDistanceCommand(
+            DriveSubsystem driveSubsystem,
+            XPropertyManager propManager,
+            RobotAssertionManager assertionManager,
+            PIDFactory pidFactory,
+            PoseSubsystem pose) {
+        
+    	this.drive = driveSubsystem;
+        this.poseSubsystem = pose;
+        this.requires(driveSubsystem);
+        this.travelManager = pidFactory.createPIDManager("Drive to position", 0.1, 0, 0, 0, 0.5, -0.5, 3, 1, 0.5);
+        headingDrivePid = pidFactory.createPIDManager("Heading module", defaultPValue, 0, 0);
+        targetHeading = new ContiguousHeading();
+    }
+    
+    /**
+     * Sets the target distance
+     * @param deltaDistance the target in inches
+     */
+    public void setDeltaDistance(double deltaDistance) {
+        this.deltaDistance = deltaDistance;
+    }
+    
+    /**
+     * Sets the target distance with a DoubleProperty
+     * @param deltaDistanceProp the DoubleProperty for the target
+     */
+    public void setDeltaDistance(DoubleProperty deltaDistanceProp) {
+        this.deltaDistanceProp = deltaDistanceProp;
+    }   
+    
+    @Override
+    public void initialize() {
+        previousPositionInches = getYDistance();
+        
+        targetHeading = poseSubsystem.getCurrentHeading();
+        
+        if (deltaDistanceProp != null) {
+            this.targetDistance = getYDistance() + deltaDistanceProp.get();
+        } else {
+            this.targetDistance = getYDistance() + deltaDistance;
+        }
+        log.info("Initializing  with distance " + targetDistance + " inches");
+    }
+
+    @Override
+    public void execute() {
+        double power = travelManager.calculate(targetDistance, getYDistance());
+        
+        leftPower = power - calculateHeadingPower();
```

Comment:
Store the result of the heading power rather than calling it again! Right now you update the PID loop twice with the same measurement, messing with the derivative and integral calculations.

(this is why I chose "reject" rather than "comment")